.github/workflows/e2e-template-validation.yaml

```diff
@@ -0,0 +1,86 @@
+name: Run Template Validation
+
+jobs:
+  uptest:
+    if: ${{ github.event.issue.pull_request }}
```

Comment:
Do we need to add a `on` block to the workflow definition similar to [this](https://github.com/upbound/upjet-provider-template/blob/f49ec6dab3700b2f3be0d55d25e777f6ac2467c4/.github/workflows/ci.yml#L8)? Or will this definition be sufficient to run workflow?